iothub/device/src/Transport/Amqp/AmqpAuthenticationRefresher.cs

```diff
@@ -50,9 +51,13 @@ public async Task InitLoopAsync(CancellationToken cancellationToken)
                     cancellationToken)
                 .ConfigureAwait(false);
 
+            // Create a linked cancellation token source which can be signaled for cancellation by both the SDK
+            // and the supplied cancellation token.
+            _refresherCancellationTokenSource = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
```

Comment:
Is this method re-entrant? If so, should we dispose before overwriting?

Reply:
The current implementation has upstream caller protection against multiple threads calling it in parallel. However, this method is defined in an interface, so I'll update it to check for an existing source and reuse it if it exists. I can't think of a scenario where this method would be called by concurrently with different cancellation tokens, so us reusing the previous token source should be fine.